spec/models/oauth_client_spec.rb

```diff
@@ -0,0 +1,62 @@
+require 'rails_helper'
+
+RSpec.describe OauthClient, type: :model do
+  def create_oauth_client(client_id, client_name, redirect_uri)
+    OauthClient.new(client_id:, client_name:, redirect_uri:)
+  end
+
+  describe 'Validations' do
```

Comment:
I would set your arguments to variables to simplify a lot of these specs that have repeated arguments. Set the valid args in the top-level context just under the describe, then you can just create the client which will work for most tests:
```ruby
let(:client_id) { '123' }
let(:client_name) { 'robert' }
let(:redirect_uri) { 'http://robert.com/callback' }
let(:oauth_client) { create_oauth_client(client_id, client_name, redirect_uri) }
```
After that you just change the individual attribute you're testing for in each test context. The "client_id has already been used" test still needs the `before` block because you're not saving the clients you create; this is an example of something that gets easier to test for with `factorybot`.